armi/reactor/grids/hexagonal.py

```diff
@@ -0,0 +1,432 @@
+from math import sqrt
+from typing import Tuple, List, Optional
+
+import numpy
+
+from armi.reactor import geometry
+from armi.utils import hexagon
+
+from .constants import (
+    BOUNDARY_0_DEGREES,
+    BOUNDARY_120_DEGREES,
+    BOUNDARY_60_DEGREES,
+    BOUNDARY_CENTER,
+)
+from .locations import IndexLocation, IJKType, IJType
+from .structuredgrid import StructuredGrid
+
+COS30 = sqrt(3) / 2.0
+SIN30 = 1.0 / 2.0
+# going CCW from "position 1" (top right)
+TRIANGLES_IN_HEXAGON = numpy.array(
+    [
+        (+COS30, SIN30),
+        (+0, 1.0),
+        (-COS30, SIN30),
+        (-COS30, -SIN30),
+        (+0, -1.0),
+        (+COS30, -SIN30),
+    ]
+)
+
+
+class HexGrid(StructuredGrid):
+    """
+    Has 6 neighbors in plane.
+
+    .. note::
```

Comment:
The way the ARMI docs are build, we use this format:

```
Notes
-----
```

The `.. note::` doesn't usually render the way we like.

I Juuuuust noticed we have 13 of these left in the repo's docstrings.  Whoops.  I will have to go fix those.